bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java

```diff
@@ -201,17 +195,23 @@ private void installPlugin(String shortname, String version) throws IOException
     }
 
     public int run() throws Throwable {
+        String appClassName = "io.jenkins.jenkinsfile.runner.App";
+        if (hasClass(appClassName)) {
+            Class<?> c = Class.forName(appClassName);
+            return ((IApp) c.newInstance()).run(this);
+        }
```

Comment:
We first check if a particular class is present in class path(in case of fat jar), otherwise the old flow will be initiated. i.e. collecting jars and then load the class